hack/update-codegen.sh

```diff
@@ -17,224 +18,7 @@ set -o errexit
-    --input-base "" \
-    --input "$joined" \
-    --trim-path-prefix="$module_name" \
-    --output-package "${client_package}"/clientset \
-    --output-base ./
-}
-
-gen-listers() {
-  clean "${client_subpackage}/listers" '*.go'
-  echo "Generating listers..." >&2
-  prefixed_inputs=( "${client_inputs[@]/#/$module_name/}" )
-  joined=$( IFS=$','; echo "${prefixed_inputs[*]}" )
-  "$listergen" \
-    --go-header-file hack/boilerplate/boilerplate.generatego.txt \
-    --input-dirs "$joined" \
-    --trim-path-prefix="$module_name" \
-    --output-package "${client_package}"/listers \
-    --output-base ./
-}
-
-gen-informers() {
-  clean "${client_subpackage}"/informers '*.go'
-  echo "Generating informers..." >&2
-  prefixed_inputs=( "${client_inputs[@]/#/$module_name/}" )
-  joined=$( IFS=$','; echo "${prefixed_inputs[*]}" )
-  "$informergen" \
-    --go-header-file hack/boilerplate/boilerplate.generatego.txt \
-    --input-dirs "$joined" \
-    --versioned-clientset-package "${client_package}"/clientset/versioned \
-    --listers-package "${client_package}"/listers \
-    --trim-path-prefix="$module_name" \
-    --output-package "${client_package}"/informers \
-    --output-base ./
-}
-
-gen-defaulters() {
-  clean internal/apis 'zz_generated.defaults.go'
-  clean pkg/webhook/handlers/testdata/apis 'zz_generated.defaults.go'
-  echo "Generating defaulting functions..." >&2
-  prefixed_inputs=( "${defaulter_inputs[@]/#/$module_name/}" )
-  joined=$( IFS=$','; echo "${prefixed_inputs[*]}" )
-  "$defaultergen" \
-    --go-header-file hack/boilerplate/boilerplate.generatego.txt \
-    --input-dirs "$joined" \
-    --trim-path-prefix="$module_name" \
-    -O zz_generated.defaults \
-    --output-base ./
-}
-
-gen-conversions() {
-  clean internal/apis 'zz_generated.conversion.go'
-  clean pkg/webhook/handlers/testdata/apis 'zz_generated.conversion.go'
-  echo "Generating conversion functions..." >&2
-
-  CONVERSION_EXTRA_PEER_PKGS=(
-    github.com/cert-manager/cert-manager/internal/apis/meta \
-    github.com/cert-manager/cert-manager/internal/apis/meta/v1 \
-    github.com/cert-manager/cert-manager/pkg/apis/meta/v1
-  )
-  CONVERSION_PKGS=( "${conversion_inputs[@]/#/$module_name/}" )
-
-  "$conversiongen" \
-      --go-header-file hack/boilerplate/boilerplate.generatego.txt \
-      --extra-peer-dirs $( IFS=$','; echo "${CONVERSION_EXTRA_PEER_PKGS[*]}" ) \
-      --extra-dirs $( IFS=$','; echo "${CONVERSION_PKGS[*]}" ) \
-      --input-dirs $( IFS=$','; echo "${CONVERSION_PKGS[*]}" ) \
-      --trim-path-prefix="$module_name" \
-      -O zz_generated.conversion \
-      --output-base ./
-}
-
-runfiles="$(pwd)"
-cd "$BUILD_WORKSPACE_DIRECTORY"
-
-gen-deepcopy
-gen-clientsets
-gen-listers
-gen-informers
-gen-defaulters
-gen-conversions
+echo "!!! hack/update-codegen.sh is now a no-op and will soon be removed" 2>&1
+echo "!!! use "make update-codegen" instead" 2>&1
 
-## Call update-bazel
-cd "$runfiles"
-"$@"
+exit 0
```

Comment:
My main focus was on the main cert-manager CI pipeline - if this calls `make update-codegen` then we end up running that twice currently since we're running both  pull-cert-manager-bazel and pull-cert-manager-make-test

I'll just make it run "make update-codegen" and the inefficiency be damned; it'll be good to at least have another reason to make us go faster ripping the bazel tests out.

Reply:
ugh that hits the issue that bazelbuild doesn't have `jq` installed so I'll have to shave that yak first... I suppose it at least is a yak that needs shaving

Reply:
I'm confused - I wouldn't expect us to continue to execute this script during the bazel build.. more just make this script call into 'make' instead of bazel

Reply:
If you remove the targets for update-codegen and verify-codegen from the BUILD.bazel file, it'll no longer attempt to run these scripts 😊 I'm just proposing keeping them in place so that as a developer, I can run update/verify by calling the script (rather than needing to learn the make incantation/modify scripts to be aware of it).

Reply:
If we're concerned that "people may assume that everything has run okay if they invoke say, verify-codegen in a CI pipeline somewhere" then to me that implies that either `bazel run //hack:verify-codegen` must fail loudly or `bazel run //hack:verify-codegen` must ultimately end up calling `make verify-codegen`.

The latter feels easier for us right now (because it minimises the amount of bazel I have to change today)

Reply:
So if we remove the targets, then it will fail loudly if a user attempts the 'bazel run' commands (and say "target does not exist").

What I'm talking about is people directly executing './hack/verify-codegen.sh', which it'd be nice if we can make continue to work (by having it call 'make verify-codegen' for the user) 😊

So basically, just removing the targets from the bazel build file and then making the update/verify scripts basically call 'make' so that it continues to work. I am less fussed about keeping the 'bazel run' invocations working if the overall intent is to remove bazel anyway...

Reply:
The only bazel changes I see as required here are to *remove* the targets altogether - unless we are misaligned on what the end goal/state is here 😄

Reply:
OK, so you're cool if we break people testing with bazel in this PR but you want to preserve `./hack/*-codegen.sh` and make them call make for you. I can do that :+1: